packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -69,145 +69,259 @@ import { applyEventListeners } from './modules/events';
 import { applyStaticClassAttribute } from './modules/static-class-attr';
 import { applyStaticStyleAttribute } from './modules/static-style-attr';
 
-export const TextHook: Hooks<VText> = {
-    create: (vnode) => {
-        const { owner } = vnode;
-
-        const elm = createText(vnode.text!);
-        linkNodeToShadow(elm, owner);
-        vnode.elm = elm;
-    },
-    update: updateNodeHook,
-    insert: insertNode,
-    move: insertNode, // same as insert for text nodes
-    remove: removeNode,
-};
-
-export const CommentHook: Hooks<VComment> = {
-    create: (vnode) => {
-        const { owner, text } = vnode;
-
-        const elm = createComment(text);
-        linkNodeToShadow(elm, owner);
-        vnode.elm = elm;
-    },
-    update: updateNodeHook,
-    insert: insertNode,
-    move: insertNode,
-    remove: removeNode,
-};
-
-// insert is called after update, which is used somewhere else (via a module)
-// to mark the vm as inserted, that means we cannot use update as the main channel
-// to rehydrate when dirty, because sometimes the element is not inserted just yet,
-// which breaks some invariants. For that reason, we have the following for any
-// Custom Element that is inserted via a template.
-export const ElementHook: Hooks<VElement> = {
-    create: (vnode) => {
-        const {
-            sel,
-            owner,
-            data: { svg },
-        } = vnode;
+export function patchChildren(c1: VNodes, c2: VNodes, parent: ParentNode): void {
+    if (hasDynamicChildren(c2)) {
+        updateDynamicChildren(c1, c2, parent);
+    } else {
+        updateStaticChildren(c1, c2, parent);
+    }
+}
+
+function patch(n1: VNode | null, n2: VNode, parent: ParentNode, anchor: Node | null) {
```

Comment:
@pmdartus I wouldn't say I like that this function is doing two different things based on the parameters. If `n1` is null, it is mounting (create+insert right?), otherwise patching. Then in the `process*`, we make the same differentiation with a condition for `isNull` (mount/patch).

imho, (because the diffing algo is already saying whether to mount/patch) a better approach will be to separate the current `patch` logic into `patch` and `mount` and the `process*` into `patch*` and `mount*` (you already have them for Element and CustomeElements)? (imo) Such a way is more explicit, where you have `patch(null,...)` will become `mount(n2,...) and we can skip the multiple `isNull` checks on the critical path of the diffing algo.

Reply:
@jodarove I went ahead and implemented this: https://github.com/salesforce/lwc/pull/2677/commits/0094a68292cb55af2c2dc519c194e7681b63599b

I'll re-run the perf tests, because maybe this actually has a perf impact? Checking `isNull()` over and over does seem like it may have a small cost.